### PR TITLE
Some miscellaneous cleanups

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Did these cleanups for a while and once I saw the DialogTitleView decided to push them. The most important(for me) is the first one which clears the project view in the IDE to not have two root packages(annoyed me quite a bit). See commits messages for more info, can squash as desired.

## How Has This Been Tested?

App still works, no build issues.

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by mikehardy — CHANGES_REQUESTED** [review]
I disagree with the Assert -> assert move, why?
Because the answer to the question "should I use platform assert()?" is always no.
The platform will never, under any conditions you are likely to execute the app on, enforce your assertions. They are just wishes, like I wish that was validated and crashed the way I thought it would, saving me from continuing to execute in an unknown state and providing me crash report data

But it won't

Our Assert class - if we truly want to assert - will actually do what you think and crash

So I think if we do anything with assert we go the other direction and all `assert` becomes our Assert

example reference but by no means unique on the subject https://stackoverflow.com/questions/6176441/how-to-use-assert-in-android

**Comment by david-allison** [comment]
@mikehardy is this still true for Kotlin?

https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/assert.html

**Comment by mikehardy** [comment]
@david-allison 

this seems important doesn't it?

> Throws an [AssertionError](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-assertion-error/index.html) if the [value](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/assert.html) is false and runtime assertions have been enabled on the JVM using the -ea JVM option.

I emphasize, *and runtime assertions have been enabled on the JVM using the -ea JVM option*

I don't trust it. I don't believe assertions are turned on in the Dalvik VM.

**Comment by lukstbit** [comment]
Interesting. 
Well there are a couple of uses of the platform assert which I see now that are meaningless. I still think our custom _Assert_ class should go. There are just a couple of uses of assert(platform or our implementation) and if the call sites should throw when the conditions are not met might as well use language constructs like if-else and more precise excptions.

**Comment by mikehardy** [comment]
> There are just a couple of uses of assert(platform or our implementation) and if the call sites should throw when the conditions are not met might as well use language constructs like if-else and more precise exceptions.

That's one way to go, and I'd be totally fine with that - this is a classic area of programmer confusion obviously :-), so that would get around the whole issue

> but if we continue with our own assert then there are some changes to be made to remove the platform method use.

Agreed with that if/then, and I'd be fine with this direction also (moving to all Assert)

So for me, either direction is fine. I guess thinking about it now and that this became a "Today I Learned..." moment, if we go with "keep our Assert..." then we better add a clearly worded explanation of why platform assert has no value but that's documenting around a mild problem. Can totally work but...

I think I lean slightly towards "remove *all* assert, replace with specific checks and IllegalStateException throws or similar as needed. I'm assuming (dangerous, could be wrong) there aren't many?

**Comment by lukstbit** [comment]
I added the code to remove all assertions from the codebase.

**Review by BrayanDSO — APPROVED** [review]
nice

**Comment by david-allison** [comment]
> CollectionDBCorruption

This was for scoped storage

I thought it might still be useful, so I didn't flag it it Brayan's huge cleanup

Happy for it to go if it's not useful

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt [hunk 1]
```diff
@@ -468,8 +467,7 @@ open class Card : Cloneable {
          * Returns [flags] with the 3 first bits set as in [flag]
          */
         fun setFlagInInt(flags: Int, flag: Int): Int {
-            Assert.that(0 <= flag, "flag to set is negative")
-            Assert.that(flag <= 7, "flag to set is greater than 7.")
+            if (flag !in 0..7) throw IllegalArgumentException("flag outside of expected [0, 7] interval")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:470]: Can we use `require { }` here, so Kotlin can apply code contracts?

It's also succinct

https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/require.html

```suggestion
            require(flag in 0..7) { "flag outside of expected [0, 7] interval" }
```

**Review by david-allison — DISMISSED** [review]
one code quality, rest looks fine

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt (hunk 1 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt]: Done!

File: AnkiDroid/src/main/java/com/ichi2/libanki/Notetypes.kt [hunk 2]
```diff
@@ -364,14 +363,18 @@ class Notetypes(val col: Collection) {
 
     @LibAnkiAlias("rename_field")
     fun renameField(notetype: NotetypeJson, field: Field, newName: String) {
-        assert(notetype.flds.jsonObjectIterable().contains(field))
+        if (!notetype.flds.jsonObjectIterable().contains(field)) {
+            throw IllegalStateException("Field to be renamed was not found in the notetype fields")
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Notetypes.kt:367]: @david-allison did you mean `check()` for all things in this commit that threw IllegalStateException (and `require()` for all the IllegalArgumentException ones) ?

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt [hunk 3]
```diff
@@ -468,8 +467,7 @@ open class Card : Cloneable {
          * Returns [flags] with the 3 first bits set as in [flag]
          */
         fun setFlagInInt(flags: Int, flag: Int): Int {
-            Assert.that(0 <= flag, "flag to set is negative")
-            Assert.that(flag <= 7, "flag to set is greater than 7.")
+            require(flag in 0..7) { "flag outside of expected [0, 7] interval" }
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt:470]: that does look nice

File: AnkiDroid/src/main/java/com/ichi2/libanki/Notetypes.kt (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Notetypes.kt:367]: I don't mind. Just require is fine

Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Notetypes.kt]: I think I got everything.

**Review by mikehardy — APPROVED** [review]
Yep, a thing of beauty, thanks
and now we all know assert() is worthless in an android context but I also learned about check() and require() in a kotlin context, so that's hopefully worth something

**Pull request merged**